Feature/responsive design

talorlangnas-moveo opened the pull request.

talorlangnas-moveo committed:
uploading chefs images

talorlangnas-moveo committed:
working on chef component

talorlangnas-moveo committed:
working on general image component

talorlangnas-moveo committed:
working on chef component

talorlangnas-moveo committed:
creating mock data file for chefImage

talorlangnas-moveo committed:
chef component ready

talorlangnas-moveo committed:
adding padding bottom

talorlangnas-moveo committed:
add Carousel component with embla-carousel integration

talorlangnas-moveo committed:
changing directories and file names to lowercase

talorlangnas-moveo committed:
Merge branch 'feature/chef' into feature/carousela

talorlangnas-moveo committed:
Add carousel styles and restaurant card data - in process

- Implemented SCSS styles for the carousel component, including layout and navigation button styles.
- Created a new TypeScript file to define and export an array of restaurant card data, including titles, descriptions, images, and ratings.

talorlangnas-moveo committed:
carousel is done

- Refactor Carousel component to accept children and update rendering logic
- add unique IDs to restaurant cards

talorlangnas-moveo committed:
Add PopularRestaurants component with styling and carousel functionality - in Process

- Created a new SCSS module for styling the PopularRestaurants component.

- Implemented the PopularRestaurants component to display popular restaurants and signature dishes using a carousel.

- Integrated restaurant cards from the data source for dynamic rendering.

talorlangnas-moveo committed:
adding type to card

talorlangnas-moveo committed:
Refactor PopularRestaurants component to accept type prop and update card styles for restaurant type

talorlangnas-moveo committed:
Add dish cards data and enhance PopularRestaurants component to display both restaurant and dish types

talorlangnas-moveo committed:
Add children prop to ChefInfo interface and update Chef component to render children if present

talorlangnas-moveo committed:
ready for pr

update ChefInfo interface to include firstName and lastName, and link chefsRestaurants to restaurantsCards in chefInfo data.

talorlangnas-moveo committed:
update components padding

talorlangnas-moveo committed:
adding breakpoints to _variables.scss and modified the icons

talorlangnas-moveo committed:
starting with hero component - in process

talorlangnas-moveo committed:
working on hero componet

Refactor hero and input search components for improved responsiveness and styling consistency.
need to improve

talorlangnas-moveo committed:
make hero component responsive

talorlangnas-moveo committed:
iconLegend component responsiveness complete

talorlangnas-moveo committed:
update and modified components

refactor: replace PopularRestaurants with InfoPanel and Carousel components; remove Chef component

- Seperate between PopularRestaurants to Carousel

- Update names for the components Chef to ChefCard, PopularRestaurants to InfoPanel

- refactor Carousel

talorlangnas-moveo committed:
deleting comments

talorlangnas-moveo committed:
refactor chefCard

- remove backgroundImage component, using Image/nextjs
- remove unused components and update image imports

talorlangnas-moveo committed:
making header component responsive

talorlangnas-moveo committed:
working on Card responsivity

talorlangnas-moveo committed:
chefCard is responsive

talorlangnas-moveo committed:
footer is responsive

talorlangnas-moveo committed:
creating dropdown component and desktop responsivity

talorlangnas-moveo committed:
Update chefCard.module.scss

talorlangnas-moveo committed:
remove comments

taluliel18 commented on epicure-app/src/components/aboutUs/AboutUs.tsx:
```diff
@@ -1,16 +1,33 @@
+                  height={52}
+                  alt="App Store Icon"
+                />
+        </div>
```
Why aren't you using the AppDownloadButtons component?

talorlangnas-moveo replied:
done

taluliel18 commented on epicure-app/src/components/card/Card.tsx:
```diff
@@ -1,25 +1,78 @@
+import { useWindowWidth } from "@/hooks/useWindowWidth";
 
 interface CardProps {
   card: CardInfo;
```
We said we'd pass the props here

talorlangnas-moveo replied:
done

taluliel18 commented on epicure-app/src/components/card/Card.tsx:
```diff
@@ -1,25 +1,78 @@
+              </div>
+            )}
+          </>
+        )}
```
Don’t check by type because that makes it non-generic.
You should check based on the props to decide whether to render something or not.
Also, perform the page width check once inside a hook and reuse the result instead of checking it repeatedly

talorlangnas-moveo replied:
done

taluliel18 commented on epicure-app/src/components/chefCard/chefCard.tsx:
```diff
@@ -0,0 +1,28 @@
+        alt="Yossi Shitrit Image"
+        sizes="100vw"
+        className={styles.chefImage}
+      />
```
You need to use this data from the outside, otherwise it’s not a generic component.
If I want to select a new chef now, you’ll be stuck with the same image and won’t be able to control the change

talorlangnas-moveo replied:
done

taluliel18 commented on epicure-app/src/components/header/Header.tsx:
```diff
@@ -1,12 +1,18 @@
     <header className={styles.header}>
-      <NavbarIcon />
-      <ForknifeIcon />
+      {width !== null && (width > 1023 ? <NavbarDesktop /> : <NavbarMobile />)}
```
Here too you're checking the width, so it's better to create a custom hook and use it wherever needed

talorlangnas-moveo replied:
done

talorlangnas-moveo committed:
modifying the code due to pr comments

taluliel18 commented on epicure-app/src/components/card/Card.tsx:
```diff
@@ -1,25 +1,83 @@
+  price,
+  logoUrl,
+}: CardInfo) {
+  const isDesktop = useBreakpoint(1023);
```
It's better to use the size inside the hook instead of passing it from everywhere,
or use a constant so that if you need to change the size in the future, you can do it from a single place rather than updating each call separately

talorlangnas-moveo replied:
done.

moved the screen width size into the hook and renamed it from useBreakpoint to useIsDesktopView.

taluliel18 commented on epicure-app/src/components/card/Card.tsx:
```diff
@@ -1,25 +1,83 @@
+            className={styles.ratingImage}
+          />
+        )}
+        {logoUrl && (
```
What is logoUrl?

talorlangnas-moveo replied:
done.

renamed logoUrl to dishCategoryLogo to better reflect its role as the icon indicating the dish's category (e.g., spicy, vegan, vegetarian).

talorlangnas-moveo committed:
refactoring useBreakpoint and logoUrl names

- replace useBreakpoint with useIsDesktopView

- replace logoUrl with dishCategoryLogo

talorlangnas-moveo committed:
Update dishCards.ts

refactor: rename logoUrl to dishCategoryLogo in dishCards data

talorlangnas-moveo committed:
remove unused SVG icons

talorlangnas-moveo committed:
Merge branch 'development' into feature/responsive-design

talorlangnas-moveo committed:
Remove unused SVG file from the project
